### PR TITLE
LibWeb: Ignore malformed at-rules in CSS parser

### LINKED ISSUE #12405: ASSERTION_FAILED using Text Editor
make a new text file, save it with a html ending, then paste this into the text box, then save:
```html
<style>
@media
</style>
```

Stack trace:
```
414.600 CrashReporter(60:62): --- Backtrace for thread #0 (TID 58) ---
414.605 CrashReporter(60:62): 0x56bea02e: [libsystem.so] syscall2 +0xe (syscall.cpp:25 => syscall.cpp:24)
414.615 CrashReporter(60:62): 0xb1705c84: [libc.so] raise +0x24 (syscall.h:35 => signal.cpp:22 => signal.cpp:37)
414.622 CrashReporter(60:62): 0xb16eab59: [libc.so] abort +0x29 (stdlib.cpp:220)
414.630 CrashReporter(60:62): 0xb16f168c: [libc.so] __assertion_failed +0x6c (assert.cpp:33)
414.630 CrashReporter(60:62): 0x50aee85e: [libweb.so.serenity] Web::CSS::Parser::convert_to_rule(AK::NonnullRefPtr<Web::
CSS::StyleRule>) [clone .localalias] +0xaae (RefPtr.h:289 => RefPtr.h:256 => StyleRule.h:30 => Parser.cpp:1942)
414.641 CrashReporter(60:62): 0x50afc5d7: [libweb.so.serenity] AK::NonnullRefPtr<Web::CSS::CSSStyleSheet> Web::CSS::Pars
er::parse_a_stylesheet<Web::CSS::Token>(Web::CSS::TokenStream<Web::CSS::Token>&) +0xa7 (Parser.cpp:174)
414.655 CrashReporter(60:62): 0x50aef40b: [libweb.so.serenity] Web::parse_css(Web::CSS::ParsingContext const&, AK::Strin
gView) +0x1bb (Parser.cpp:164 => Parser.cpp:4527)
414.664 CrashReporter(60:62): 0x50bc6123: [libweb.so.serenity] Web::HTML::HTMLStyleElement::update_a_style_block() [clon
e .localalias] +0x1b3 (HTMLStyleElement.cpp:125)
414.674 CrashReporter(60:62): 0x50bc663b: [libweb.so.serenity] Web::HTML::HTMLStyleElement::children_changed() [clone .l
ocalalias] +0x1b (HTMLStyleElement.cpp:26)
414.680 CrashReporter(60:62): 0x50bd20bc: [libweb.so.serenity] Web::HTML::HTMLParser::flush_character_insertions() [clon
e .part.0] +0x8c (HTMLParser.cpp:835)
414.694 CrashReporter(60:62): 0x50bec7d1: [libweb.so.serenity] Web::HTML::HTMLParser::run(AK::URL const&) +0x991 (HTMLPa
rser.cpp:832 => HTMLParser.cpp:187)
414.700 CrashReporter(60:62): 0x50c4e20d: [libweb.so.serenity] Web::FrameLoader::load_html(AK::StringView, AK::URL const
&) +0xbd (FrameLoader.cpp:217)
414.708 CrashReporter(60:62): 0x50c7763d: [libweb.so.serenity] Web::Page::load_html(AK::StringView, AK::URL const&) +0x2
d (Page.cpp:46)
414.713 CrashReporter(60:62): 0xa8911358: [/bin/WebContent] .L4418 +0x68 (ClientConnection.cpp:94 => WebContentServerEnd
point.h:2459)
414.725 CrashReporter(60:62): 0x0d9e6612: [libipc.so.serenity] IPC::ConnectionBase::handle_messages() [clone .localalias
] +0x102 (Connection.cpp:95)
414.730 CrashReporter(60:62): 0x3f12f795: [libcore.so.serenity] AK::Function<void ()>::CallableWrapper<Core::Object::def
erred_invoke(AK::Function<void ()>)::{lambda()#1}>::call() +0x245 (Function.h:91)
414.743 CrashReporter(60:62): 0x3f117923: [libcore.so.serenity] Core::EventLoop::pump(Core::EventLoop::WaitMode) [clone
.localalias] +0x3a3 (Function.h:91)
414.748 CrashReporter(60:62): 0x3f11836a: [libcore.so.serenity] Core::EventLoop::exec() +0x8a (EventLoop.cpp:409)
414.759 CrashReporter(60:62): 0xa89144af: [/bin/WebContent] serenity_main(Main::Arguments) +0x1ef (main.cpp:26)
414.759 CrashReporter(60:62): 0xa760e1e7: [libmain.so.serenity] main +0x77 (Main.cpp:23)
414.766 CrashReporter(60:62): 0xa88fbd15: [/bin/WebContent] _entry +0x65 (crt0.cpp:49)
```

### PR DESCRIPTION
**Pull request by eggpi**
Fixes #12405.

### CODE REVIEW

File: Userland/Libraries/LibWeb/CSS/Parser/Parser.cpp [hunk 1]
```diff
@@ -1934,8 +1934,14 @@ RefPtr<CSSRule> Parser::convert_to_rule(NonnullRefPtr<StyleRule> rule)
 
             auto media_query_tokens = TokenStream { rule->prelude() };
             auto media_query_list = parse_a_media_query_list(media_query_tokens);
+            if (!media_query_list.size()) {
+                return {};
+            }
```
Comment (AtkinsSJ) [inline Userland/Libraries/LibWeb/CSS/Parser/Parser.cpp:1939]: `if (media_query_list.is_empty())` would be clearer here. :^) Also, please remove the braces when it's a single line.

File: Userland/Libraries/LibWeb/CSS/Parser/Parser.cpp [hunk 2]
```diff
@@ -1934,8 +1934,14 @@ RefPtr<CSSRule> Parser::convert_to_rule(NonnullRefPtr<StyleRule> rule)
...
 
-            auto child_tokens = TokenStream { rule->block().values() };
+            if (!rule->block()) {
+                return {};
+            }
```
Comment (AtkinsSJ) [inline Userland/Libraries/LibWeb/CSS/Parser/Parser.cpp:1943]: Remove the braces here too. Could even combine the two `if`s if you want.

File: Userland/Libraries/LibWeb/CSS/Parser/Parser.cpp [hunk 3]
```diff
@@ -1980,7 +1986,10 @@ RefPtr<CSSRule> Parser::convert_to_rule(NonnullRefPtr<StyleRule> rule)
                 return {};
             }
 
-            auto child_tokens = TokenStream { rule->block().values() };
+            if (!rule->block()) {
+                return {};
+            }
```
Comment (AtkinsSJ) [inline Userland/Libraries/LibWeb/CSS/Parser/Parser.cpp:1991]: Can remove the {} here too.

**Review by AtkinsSJ — CHANGES_REQUESTED** [review]
Nice work! :^)

I have a few nit-picks but nothing major.

**Comment by eggpi** [comment]
All done, thanks for the quick review! Please take another look.

**Review by AtkinsSJ — APPROVED** [review]
LGTM!

**Pull request merged**